Handle limited import of array of struct/map of struct/map of primitives

toregge opened the pull request:
@geirst: please review

Commit:
Handle import of array of struct fields, map of struct fields, and map of
primitive fields as long as required attributes are present.

geirst commented on config-model/src/main/java/com/yahoo/searchdefinition/derived/ImportedFields.java:
```diff
@@ -39,8 +40,17 @@ public void getConfig(ImportedFieldsConfig.Builder builder) {
-            builder.attribute.add(createAttributeBuilder(field));
+        ImmutableSDField targetField = field.targetField();
+        String targetFieldName = targetField.getName();
+        if (targetFieldName.indexOf('.') == -1) {
```
Consider adding a helper function (!isNestedFieldName(...)) to better describe what we try to do here.

geirst commented on config-model/src/test/java/com/yahoo/searchdefinition/processing/ImportedFieldsTestCase.java:
```diff
@@ -65,6 +69,173 @@ private static Search buildAdSearch(String sdContent) throws ParseException {
+    }
+
+    @Test
+    public void check_illegal_stroct_import_missing_array_of_struct_attributes() throws ParseException {
```
Typo: _stroct_ -> _struct_

geirst commented on config-model/src/test/java/com/yahoo/searchdefinition/processing/ImportedFieldsTestCase.java:
```diff
@@ -65,6 +69,173 @@ private static Search buildAdSearch(String sdContent) throws ParseException {
+            List<String> result = new ArrayList<String>();
+            if (isAttribute) {
+                result.add("        indexing: attribute");
+                if (isFastSearch) {
```
Is the fast-search aspect really needed / used in these tests? If not, consider removing it to simplify.

geirst commented on config-model/src/test/java/com/yahoo/searchdefinition/processing/ImportedFieldsTestCase.java:
```diff
@@ -65,6 +69,173 @@ private static Search buildAdSearch(String sdContent) throws ParseException {
         return builder.getSearch("ad");
     }
 
+    private static void check_struct_import(ParentSdBuilder parentBuilder) throws ParseException {
```
Please rename to checkStructImport. Underscores are mainly for test function names.

geirst reviewed (commented):
Overall looks very good, please have a look at my comments.

Commit:
Fix typo.
Factor out logic to check for nested field name.
Use camel case instead of underscore for compound function name.
Simplify test.

geirst reviewed (approved):
👍
